src/rules/jsxAlignmentRule.ts

```diff
@@ -68,20 +68,21 @@ function walk(ctx: Lint.WalkContext<void>) {
     function checkElement(
         elementOpen: ts.LineAndCharacter,
         attributes: Array<ts.JsxAttribute | ts.JsxSpreadAttribute> // TS <=2.2
-            | { properties: Array<ts.JsxAttribute | ts.JsxSpreadAttribute> }, // TS 2.3
+            | { properties: Array<ts.JsxAttribute | ts.JsxSpreadAttribute> } // TS 2.3
+            | ts.JsxAttributes, // TS 2.6
         elementClose: ts.LineAndCharacter,
         closingTag?: ts.JsxClosingElement,
     ) {
-        attributes = Array.isArray(attributes) ? attributes : attributes.properties;
-        if (attributes.length === 0) {
+        const attrs = Array.isArray(attributes) ? attributes : attributes.properties;
```

Comment:
TS 2.6.2 did not like the reassignment of attributes.  Setting to a new variable and adding the typing resolved the type checks.